lib/Runtime/Library/JavascriptString.cpp

```diff
@@ -2128,23 +2110,21 @@ namespace Js
         wchar_t *outStr = builder.DangerousGetWritableBuffer();
 
         wchar_t* outStrLim = outStr + count;
+        wchar_t *o = outStr;
+
+        while (o < outStrLim)
+        {
+            *o++ = *inStr++;
+        }
 
-        //TODO the mapping based on unicode tables needs review for ES5.
-        // mapping is not always 1.1
         if(toCase == ToUpper)
         {
-            while (outStr < outStrLim)
-            {
-                *outStr++ = CharToUpperCase(*inStr++);
-            }
+            CharUpperBuffW(outStr, count);
         }
         else
         {
             Assert(toCase == ToLower);
-            while(outStr < outStrLim)
-            {
-                *outStr++ = CharToLowerCase(*inStr++);
-            }
+            CharLowerBuffW(outStr, count);
```

Comment:
I didn't find good documentation of `CharLowerBuffW` and `CharUpperBuffW`, but looking at [CharUpperBuff](https://msdn.microsoft.com/en-us/library/windows/desktop/ms647475(v=vs.85).aspx), it seems that it returns no. of characters processed. Can we have an assert on it that it matches the count? Unless it returns no. of characters it converted to in which case it won't match for characters listed in [SpecialCasing.txt](http://www.unicode.org/Public/5.1.0/ucd/SpecialCasing.txt).